mlrun/runtimes/base.py

```diff
@@ -1148,42 +1271,58 @@ def _ensure_runtime_resource_run_logs_collected(
                 store_log = True
 
         if store_log:
-            logger.info("Storing runtime resource log before deletion")
-            logs_from_k8s, _ = crud.Logs.get_log(
+            _, logs_from_k8s = crud.Logs.get_log(
                 db_session, project, uid, source=LogSources.K8S
             )
-            crud.Logs.store_log(logs_from_k8s, project, uid, append=False)
+            if logs_from_k8s:
+                logger.info("Storing finished run logs", project=project, uid=uid)
+                crud.Logs.store_log(logs_from_k8s, project, uid, append=False)
 
     @staticmethod
-    def _ensure_runtime_resource_run_status_updated(
+    def _ensure_run_status_updated(
```

Comment:
```suggestion
    def _update_run_state(
```
Also, isn't there some "Run" class that you can implement this as  `run.update_state(...)` on ?

Reply:
The Run class is for the SDK, no such a thing for the API